kernel/term_typing.ml

```diff
@@ -539,8 +539,8 @@ let translate_local_assum env t =
 let translate_recipe env kn r =
   (** We only hashcons the term when outside of a section, otherwise this would
       be useless. It is detected by the dirpath of the constant being empty. *)
-  let (_, dir, _) = Constant.repr3 kn in
-  let hcons = DirPath.is_empty dir in
+(*  let (_, dir, _) = Constant.repr3 kn in *)
+  let hcons = false (* XXX DirPath.is_empty dir *) in
```

Comment:
Isn't that preventing section definitions from being hashconsed when they exit it?

Reply:
In fact now that we don't have to change the names I think we could just hashcons when the `cooking_info` is empty.

Reply:
Indeed.

Reply:
> Isn't that preventing section definitions from being hashconsed when they exit it?

Yeah, I completely forgot to come back to this part of the code, sorry.

Reply:
> Isn't that preventing section definitions from being hashconsed when they exit it?

In fact, I don't think so because the only call to `translate_recipe` already hash conses everything (body, type, universes)...

Can you confirm?